### PR TITLE
Allow __call metatable backed module in hotkey binds

### PR DESCRIPTION
**Pull request by junkblocker**
Implements #932 .

### CODE REVIEW

File: extensions/hotkey/init.lua [hunk 1]
```diff
@@ -143,6 +143,19 @@ local function getIndex(mods,keycode) -- key for hotkeys table
...
   return mods..key
 end
+
+local function getFunc(f)
+  if f == nil then return nil end
+  if type(f) == 'function' then return f end
+  if type(f) == 'table' then
+    local m = getmetatable(f)
+    if m and m.__call then
+      return getFunc(m.__call)
```
Comment (cmsj) [inline extensions/hotkey/init.lua]: This is a very edge case, but this is potentially infinitely recursive if someone has a weird/broken table. How about just changing line 152/153 to:

```lua
if m and m.__call and type(m.__call) == 'function' then
  return m.__call
```

?

File: extensions/hotkey/init.lua [hunk 2]
```diff
@@ -176,23 +189,26 @@ function hotkey.new(mods, key, message, pressedfn, releasedfn, repeatfn)
...
   -- message can be omitted
-  if message==nil or type(message)=='function' then
+  if message==nil or getFunc(message) then
     repeatfn=releasedfn releasedfn=pressedfn pressedfn=message message=nil -- shift down arguments
   end
-  if type(pressedfn)~='function' and type(releasedfn)~='function' and type(repeatfn)~='function' then
+  local f_pressedfn = getFunc(pressedfn)
+  local f_releasedfn = getFunc(releasedfn)
+  local f_repeatfn = getFunc(repeatfn)
+  if not f_pressedfn and not f_releasedfn and not f_repeatfn then
```
Comment (cmsj) [inline extensions/hotkey/init.lua]: I think I'd be happier here if 195-197 were:

```lua
pressedfn = getFunc(pressedfn)
releasedfn = getFunc(releasedfn)
repeatfn = getFunc(repeatfn)
```

to reduce the size of the rest of the patch, and eliminate any confusion about which variables are current/correct.

**Comment by cmsj** [comment]
I added a couple of notes, but I'm happy to make the changes myself :)

**Comment by junkblocker** [comment]
Thanks, @cmsj ! :)

**Pull request closed without merging**